iroha/src/validation.rs

```diff
@@ -13,9 +13,15 @@ pub struct MerkleTree {
 }
 
 impl MerkleTree {
+    pub fn new() -> Self {
+        MerkleTree {
+            root_node: Node::Empty,
+        }
+    }
+
     /// Builds a Merkle Tree from sorted array of `Blocks`.
     //TODO: should we check or sort blocks here?
-    pub fn build(blocks: &[&Block]) -> Self {
+    pub fn build(&mut self, blocks: &[&Block]) {
```

Comment:
Haven't noticed before, but I think we should change `&[&Block]` type. Why use slice of references if it's itself a reference? I think it unnecessarily complicates this type.

Reply:
@eadventurous I assume that right now we don't have guarantees about mutability of blocks passed between different modules and using slice of references I try to protect API from algorithmic [connascence](https://en.wikipedia.org/wiki/Connascence).
(More about the term: https://www.youtube.com/watch?v=fSr8LDcb0Y0)